pkg/quota/admission/clusterresourceoverride/admission.go

```diff
@@ -251,7 +260,61 @@ func updateContainerResources(config *internalConfig, container *kapi.Container)
 		if cpuFloor.Cmp(*q) > 0 {
 			q = cpuFloor.Copy()
 		}
+		if nsCPUFloor != nil && q.Cmp(*nsCPUFloor) < 0 {
+			glog.V(5).Infof("%s: %s pod limit %q below namespace limit; setting limit to %q", api.PluginName, kapi.ResourceCPU, q.String(), nsCPUFloor.String())
+			q = nsCPUFloor.Copy()
+		}
 		resources.Requests[kapi.ResourceCPU] = *q
 	}
+}
+
+// minResourceLimits finds the minimum CPU and minimum Memory resource
+// values across all the limits in limitRanges. Nil is returned if
+// there is CPU or Memory resource respectively.
+func minResourceLimits(limitRanges []*kapi.LimitRange) (*resource.Quantity, *resource.Quantity) {
+	cpuLimits := []*resource.Quantity{}
+	memLimits := []*resource.Quantity{}
+
+	for _, limitRange := range limitRanges {
+		for _, limit := range limitRange.Spec.Limits {
+			if limit.Type != kapi.LimitTypeContainer {
```

Comment:
I'm not familiar with what this means... if there's a pod limitrange that says min is 256MB, and a container limit range that says the min is 384MB, does that not mean the min on a container is 384? if so, don't we need to include this type of limit?

I also expected opting into specifically recognized types (pod and maybe container), rather than negative matching (this logic would include PV limit ranges, for example)

Reply:
> I also expected opting into specifically recognized types (pod and maybe container), rather than negative matching (this logic would include PVC limit ranges, for example)

Doesn't this only consider container types?

There's an explicit test case for checking against non-container types:
  https://github.com/openshift/origin/pull/18553/commits/e76cb4d38b0ded3f9d70a4cc712b8977060f26b2#diff-62f7f5e71423c1d5dfa6407e3984804bR306

Happy to drop the `continue` and do the appends iff it's a container type.

Reply:
This looks fine to me.   You are only modifying pod objects and not pvcs in this scenario.  Checking for limit type of Container only is fine.